compiler-core/src/analyse/imports.rs

```diff
@@ -242,28 +224,72 @@ impl<'a> Importer<'a> {
         Ok(())
     }
 
-    fn register_module_usage(&mut self, import: &Import<()>) {
-        if import.unqualified_types.is_empty() && import.unqualified_values.is_empty() {
-            // When the module has no unqualified imports, we track its usage
-            // so we can warn if not used by the end of the type checking
-            let _ = self
-                .environment
-                .unused_modules
-                .insert(import.used_name(), import.location);
-        } else if let Some(as_name) = &import.as_name {
-            // When the module has a name, we also track its as_name usage
-            // so we can warn if not used by the end of the type checking
-            let _ = self
-                .environment
-                .unused_modules
-                .insert(as_name.name.clone(), as_name.location);
-            // We also register it's name to differentiate between unused module
-            // and unused module name. See 'convert_unused_to_warnings'.
-            let _ = self
-                .environment
-                .imported_module_aliases
-                .insert(as_name.name.clone());
-        }
+    fn register_module(
+        &mut self,
+        import: &Import<()>,
+        import_info: &'a ModuleInterface,
+    ) -> Result<(), Error> {
+        match import.clone().as_name {
+            Some((AssignName::Variable(used_name), location)) => {
+                self.check_not_a_duplicate_import(&used_name, import.location)?;
+
+                if import.unqualified_types.is_empty() && import.unqualified_values.is_empty() {
+                    // When the module has no unqualified imports, we track its usage
+                    // so we can warn if not used by the end of the type checking
+                    let _ = self
+                        .environment
+                        .unused_modules
+                        .insert(used_name.clone(), import.location);
+                }
```

Comment:
I think this check is exactly the same below. Could we move these out to a shared block please?

```rust
if let Some(name) = import.used_name() {
    self.check_not_a_duplicate_import(&used_name, import.location)?;

    self.environment.imported_modules.insert(name, ...)

    if !import.has_any_unqualified() {
        ...
    }
}
```